tools/src/test/java/org/apache/kafka/tools/consumer/group/DescribeConsumerGroupTest.java

```diff
@@ -141,699 +186,857 @@ public void testPrintVersion(String quorum) {
+                assertEquals(TimeoutException.class, e.getCause().getClass());
+            }
+        }
     }
 
-    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
-    @MethodSource({"getTestQuorumAndGroupProtocolParametersAll"})
-    public void testDescribeGroupOffsetsWithShortInitializationTimeout(String quorum, String groupProtocol) {
-        // Let creation of the offsets topic happen during group initialization to ensure that initialization doesn't
-        // complete before the timeout expires
+    @ClusterTemplate("generator")
+    public void testDescribeGroupStateWithShortInitializationTimeout() throws Exception {
+        for (GroupProtocol groupProtocol: clusterInstance.supportedGroupProtocols()) {
+            String topic = TOPIC_PREFIX + groupProtocol.name();
+            String group = GROUP_PREFIX + groupProtocol.name();
+            createTopic(topic);
 
-        // run one consumer in the group consuming from a single-partition topic
-        addConsumerGroupExecutor(1, groupProtocol);
+            // Let creation of the offsets topic happen during group initialization to ensure that initialization doesn't
+            // complete before the timeout expires
 
-        // set the group initialization timeout too low for the group to stabilize
-        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--group", GROUP, "--timeout", "1"};
-        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+            // run one consumer in the group consuming from a single-partition topic
+            try (AutoCloseable protocolConsumerGroupExecutor = consumerGroupClosable(groupProtocol, group, topic, Collections.emptyMap());
+                 // set the group initialization timeout too low for the group to stabilize
+                 ConsumerGroupCommand.ConsumerGroupService service = consumerGroupService(new String[]{"--bootstrap-server", clusterInstance.bootstrapServers(), "--describe", "--group", group, "--timeout", "1"});
+            ) {
+                Throwable e = assertThrows(ExecutionException.class, () -> service.collectGroupState(group));
+                assertEquals(TimeoutException.class, e.getCause().getClass());
+            }
+        }
+    }
 
-        Throwable e = assertThrows(ExecutionException.class, () -> service.collectGroupOffsets(GROUP));
-        assertEquals(TimeoutException.class, e.getCause().getClass());
+    @ClusterTemplate("generator")
+    public void testDescribeWithUnrecognizedNewConsumerOption() {
+        String group = GROUP_PREFIX +  "unrecognized";
+        String[] cgcArgs = new String[]{"--new-consumer", "--bootstrap-server", clusterInstance.bootstrapServers(), "--describe", "--group", group};
+        assertThrows(joptsimple.OptionException.class, () -> consumerGroupService(cgcArgs));
```

Comment:
Could we verify `ConsumerGroupCommandOptions.fromArgs(args)` instead of creating service?